server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1601,10 +1607,16 @@ public Tuple<GatedCloseable<SegmentInfos>, ReplicationCheckpoint> getLatestSegme
                         getEngine().config().getCodec().getName()
                     )
                 );
-            } catch (IOException e) {
-                throw new OpenSearchException("Error Fetching SegmentInfos and latest checkpoint", e);
+            } catch (IOException | AlreadyClosedException e) {
+                logger.error("Error Fetching SegmentInfos and latest checkpoint", e);
+                try {
+                    snapshot.close();
+                } catch (IOException ex) {
+                    logger.error("Error while closing segmentInfosSnapshot", ex);
```

Comment:
This could indicate a larger issue, thinking we should rethrow as UncheckedIOException?